demos/python_demos/machine_translation_demo/machine_translation_demo.py

```diff
@@ -206,11 +208,14 @@ def main(args):
         tokenizer_tgt=args.tokenizer_tgt,
         output_name=args.output_name
     )
+    inputs = iter(args.input) if args.input else None
```

Comment:
IMO, the input reading logic should be abstracted out by doing something like this:

```python
if args.input:
    def iter_sentences():
        for sentence in args.input:
            print(">", sentence)
            yield sentence
else:
    def iter_sentences():
        logger.info("enter empty string to exit")
        return iter(lambda: input("> "), '')
```

And then the main loop can just be `for sentence in iter_sentences()`.

Reply:
Done. Thanks for great idea!